src/webhooks/models/webhooksFormatter.js

```diff
@@ -0,0 +1,128 @@
+        ` this is ${resultText} the threshold of ${benchmarkThreshold}. ${lastScores.length > 0 ? `last 3 scores are: ${lastScores.join()}` : 'no last score to show'}` +
+        `.*\n${statsFormatter.getStatsFormatted('aggregate', aggregatedReport.aggregate, { score })}\n`;
+}
+
+function slackWebhookFormat(message, options) {
+    return {
+        text: message,
+        icon_emoji: options.icon || WEBHOOK_SLACK_DEFAULT_MESSAGE_ICON,
+        username: WEBHOOK_SLACK_DEFAULT_REPORTER_NAME
+    };
+}
+
+function json(event, testId, jobId, report, additionalInfo, options) {
+    let payload = {
+        test_id: testId,
+        job_id: jobId,
+        event: event,
+        additional_details: {
+            ...cloneDeep({ report, ...additionalInfo })
+        }
+    };
+    return payload;
+}
+
+function slack(event, testId, jobId, report, additionalInfo, options) {
+    let message = null;
+    const {
+        environment,
+        duration,
+        parallelism = 1,
+        ramp_to: rampTo,
+        arrival_rate: arrivalRate,
+        test_name: testName,
+        grafana_report: grafanaReport
+    } = report;
+    const { score, aggregatedReport, reportBenchmark, benchmarkThreshold, lastScores, stats } = additionalInfo;
+    switch (event) {
+        case WEBHOOK_EVENT_TYPE_STARTED: {
+            let rampToMessage = rampTo ? `ramp to: ${rampTo} scenarios per second` : '';
+            message = `🤓 *Test ${testName} with id: ${testId} has started*.\n
+            *test configuration:* environment: ${environment} duration: ${duration} seconds, arrival rate: ${arrivalRate} scenarios per second, number of runners: ${parallelism}, ${rampToMessage}`;
+            break;
+        }
+        case WEBHOOK_EVENT_TYPE_FINISHED: {
+            message = `😎 *Test ${testName} with id: ${testId} is finished.*\n ${statsFormatter.getStatsFormatted('aggregate', aggregatedReport.aggregate, reportBenchmark)}\n`;
+            break;
+        }
+        case WEBHOOK_EVENT_TYPE_FAILED: {
+            message = `😞 *Test with id: ${testId} Failed*.\n
+            test configuration:\n
+            environment: ${environment}\n
+            ${stats.data}`;
+            break;
+        }
+        case WEBHOOK_EVENT_TYPE_ABORTED: {
+            message = `😢 *Test ${testName} with id: ${testId} was aborted.*\n`;
+            break;
+        }
+        case WEBHOOK_EVENT_TYPE_BENCHMARK_FAILED:
+        case WEBHOOK_EVENT_TYPE_BENCHMARK_PASSED: {
+            message = getThresholdSlackMessage(event, { testName, lastScores, aggregatedReport, benchmarkThreshold, score });
+            break;
+        }
+        case WEBHOOK_EVENT_TYPE_IN_PROGRESS: {
+            message = `:hammer_and_wrench: *Test ${testName} with id: ${testId} is in progress!*`;
+            break;
+        }
+        case WEBHOOK_EVENT_TYPE_API_FAILURE: {
+            message = `::boom:: *Test ${testName} with id: ${testId} has encountered an API failure!* :skull:`;
```

Comment:
can add in the msg the additionalInfo sent when firing the WEBHOOK_EVENT_TYPE_API_FAILURE 
 webhook with txt regarding the statusCodes > 500: `accumulatedStatusCodesCounter`